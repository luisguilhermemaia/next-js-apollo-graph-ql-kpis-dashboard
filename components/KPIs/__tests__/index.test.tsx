import '@testing-library/jest-dom/extend-expect';
import { RenderResult, render } from '@testing-library/react';
import React from 'react';
import { KPIs } from '../index';

describe('KPIs', () => {
  let renderResult: RenderResult;
  const data = {
    id: '4a6134be-007a-4a0b-9d3e-11885a40c5e9',
    activeSource: {
      id: '45803e2f-f109-4299-bdf7-66106b01f7cf',
      currentPeriod: 92500,
      lastPeriod: 1127,
    },
    weeklyActive: {
      id: '65154554-7b6a-4d1d-9691-4609ecc059fe',
      currentPeriod: 25524,
      lastPeriod: 95351,
    },
    nps: {
      id: 'e8dc4f9a-17e8-4ec9-9a46-5bb0a31f162d',
      currentPeriod: 60124,
      lastPeriod: 24545,
      avg: 67933,
    },
  };

  const doRender = () => {
    renderResult = render(
      <KPIs
        loading={false}
        activeSource={data.activeSource}
        nps={data.nps}
        weeklyActive={data.weeklyActive}
      />
    );
  };

  it('renders correctly', () => {
    doRender();

    const { container } = renderResult;

    expect(container).toMatchSnapshot();
  });
});
