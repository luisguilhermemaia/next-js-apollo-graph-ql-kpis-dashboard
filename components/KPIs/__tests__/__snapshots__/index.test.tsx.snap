// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KPIs renders correctly 1`] = `
<div>
  <div
    class="KPIs__StyledKPIsContainer-nsca2n-2 hkSEZn"
  >
    <div
      class="KPIs__StyledKPICard-nsca2n-3 KPIs__ActiveSourceContainer-nsca2n-4 kSBhyj bAwyEF"
    >
      <div>
        <p>
          Active Sourcing
        </p>
        <h3>
          92500
        </h3>
        <p>
          <span>
            1127
          </span>
           
          Last period
        </p>
      </div>
      <div
        class="KPIs__BottomCircle-nsca2n-0 cYzzdY"
      />
      <div
        class="KPIs__TopCircle-nsca2n-1 fIiNhC"
      />
    </div>
    <div
      class="KPIs__StyledKPICard-nsca2n-3 KPIs__WeeklyActiveContainer-nsca2n-5 kSBhyj jKBzOQ"
    >
      <div>
        <p>
          Weekly Active
        </p>
        <h3>
          25524
        </h3>
        <p>
          <span>
            95351
          </span>
           
          Last period
        </p>
      </div>
      <div
        class="KPIs__BottomCircle-nsca2n-0 cYzzdY"
      />
      <div
        class="KPIs__TopCircle-nsca2n-1 fIiNhC"
      />
    </div>
    <div
      class="KPIs__StyledKPICard-nsca2n-3 KPIs__NpsContainer-nsca2n-6 kSBhyj mPsyW"
    >
      <div>
        <p>
          NPS
        </p>
        <h3>
          60124
        </h3>
        <p>
          <span>
            24545
          </span>
           
          Last period
        </p>
      </div>
      <div
        class="KPIs__BottomCircle-nsca2n-0 cYzzdY"
      />
      <div
        class="KPIs__TopCircle-nsca2n-1 fIiNhC"
      />
    </div>
  </div>
</div>
`;
